server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts

```diff
@@ -53,49 +50,60 @@ clinicalDetailsRoute.post('/saveClinicalDetails', (request: Request, response: R
         wasHospitalized: clinicalDetails.wasHospitalized
     }
 
-    const requestAddress: Address = {
-        cityValue: isolationAddress?.city,
-        streetValue: isolationAddress?.street,
-        floorValue: +isolationAddress?.floor,
-        houseNumValue: +isolationAddress?.houseNum,
-    }
-    
-    graphqlRequest(CREATE_ISOLATION_ADDRESS, {
-        input: { ...requestAddress }
-    }).then((result: CreateAddressResponse) => {
-        graphqlRequest(UPDATE_INVESTIGATION, {
-            epidemiologyNumber: requestInvestigation.epidemiologyNumber,
-            hospital: requestInvestigation.hospital,
-            hospitalizationEndTime: requestInvestigation.hospitalizationEndTime,
-            hospitalizationStartTime: requestInvestigation.hospitalizationStartTime,
-            isInIsolation: requestInvestigation.isInIsolation,
-            isIsolationProblem: requestInvestigation.isIsolationProblem,
-            isIsolationProblemMoreInfo: requestInvestigation.isIsolationProblemMoreInfo,
-            isolationEndTime: requestInvestigation.isolationEndTime,
-            isolationStartTime: requestInvestigation.isolationStartTime,
-            symptomsStartTime: requestInvestigation.symptomsStartTime,
-            wasHospitalized: requestInvestigation.wasHospitalized,
-            doesHaveSymptoms: requestInvestigation.doesHaveSymptoms,
-            isolationAddress: result.data.insertAndGetAddressId.integer
+    graphqlRequest(UPDATE_INVESTIGATION, request.headers,{
+        epidemiologyNumber: requestInvestigation.epidemiologyNumber,
```

Comment:
Im not fimiliar with the investigation interface/objects, but wouldn't it be more comfortable/readable composing an object dedicated for sending an investigation to update? correct me if im missing something

Reply:
After checking it I used only one var as you can see